internal/api/handler.go

```diff
@@ -1,45 +1,59 @@
 package api
 
 import (
-	"net/http"
-
+	pb "github.com/alexdunne/gs-onboarding/internal/api/protobufs"
 	"github.com/alexdunne/gs-onboarding/internal/database"
-	"github.com/labstack/echo/v4"
+	"github.com/alexdunne/gs-onboarding/internal/models"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 type Handler struct {
-	DB database.Database
+	pb.UnimplementedAPIServer
+	Reader database.ItemReader
 }
 
-func (h *Handler) HandleGetAllItems(c echo.Context) error {
-	items, err := h.DB.GetAll(c.Request().Context())
+func (h Handler) ListAll(empty *emptypb.Empty, s pb.API_ListAllServer) error {
```

Comment:
Do you need the empty?

Reply:
Yeah the method we're implementing are provided by the [UnimplementedAPIServer](https://github.com/alexdunne/gs-onboarding/pull/4/files/f375dd1097c0f166c404fdbb1e140836fbb4f0df#diff-82546aaa7f5dddfb032d0f6f7766366895ef317fb4487ac376e256512f4e6ef4R142) which we embed in the `Handler`. [ListAll example](https://github.com/alexdunne/gs-onboarding/pull/4/files/f375dd1097c0f166c404fdbb1e140836fbb4f0df#diff-82546aaa7f5dddfb032d0f6f7766366895ef317fb4487ac376e256512f4e6ef4R145)